scripts/ci/publish.ts

```diff
@@ -437,43 +420,38 @@ function getMaxPatchVersionIncrement(versions: string[]): number {
   return Math.max(...increments, 0)
 }
 
-// TODO: we can simplify this into one line (cc Joël)
-async function getAllVersions(packages: Packages, channel: string, prefix: string): Promise<string[]> {
-  return unique(
-    flatten(
-      await pMap(
-        Object.values(packages).filter((p) => p.name !== '@prisma/integration-tests'),
-        async (pkg) => {
-          if (pkg.name === '@prisma/integration-tests') {
-            return []
-          }
-          const pkgVersions = [] as string[]
-          if (pkg.version.startsWith(prefix)) {
-            pkgVersions.push(pkg.version)
-          }
-          const remoteVersionsString = await runResult('.', `npm info ${pkg.name} versions --json`)
-
-          const remoteVersions = JSON.parse(remoteVersionsString)
-
-          for (const remoteVersion of remoteVersions) {
-            if (
-              remoteVersion.includes(channel) &&
-              remoteVersion.startsWith(prefix) &&
-              !pkgVersions.includes(remoteVersion)
-            ) {
-              pkgVersions.push(remoteVersion)
-            }
-          }
-          return pkgVersions
-        },
-        { concurrency: 3 }, // Let's not spam npm too much
-      ),
-    ),
-  )
+export async function getLatestVersionPublishedFor(pkgs: Packages, channel: string, prefix: string): Promise<string[]> {
+  // We check the versions for the `@prisma/debug` package
+  // Why?
+  // Because `@prisma/debug` is the first package that will be published
+  // So if npm fails to publish one of the packages,
+  // we cannot republish on the same version on a next run
+  const pkg = pkgs['@prisma/debug']
+
+  const values = async (pkg: Package) => {
+    const pkgVersions = [] as string[]
+    if (pkg.version.startsWith(prefix)) {
+      pkgVersions.push(pkg.version)
+    }
+
+    const remoteVersionsString = await runResult('.', `npm info ${pkg.name} versions --json`)
+    const remoteVersions: string = JSON.parse(remoteVersionsString)
+
+    for (const remoteVersion of remoteVersions) {
+      if (remoteVersion.includes(channel) && remoteVersion.startsWith(prefix) && !pkgVersions.includes(remoteVersion)) {
+        pkgVersions.push(remoteVersion)
+      }
+    }
+
+    return pkgVersions
+  }
+
+  return [...new Set(await values(pkg))]
 }
 
 async function getNextMinorStable() {
-  const remoteVersion = await runResult('.', `npm info prisma version`)
+  // We check the Prisma CLI `latest` version
+  const remoteVersion = await runResult('.', 'npm info prisma version')
```

Comment:
should it be `npm info @prisma/debug` instead?

Reply:
Here we get the version for `latest`, I think currently it makes more sense to check for `@prisma/client` or `prisma` CLI

Reply:
Just trying to understand. I realize my question above actually apply here. We start doing a `4.10` release, `@prisma/debug` goes out and then the rest fails. If I check `@prisma/client`, the next minor is `4.10`? Or did I get it wrong again :smile:

Reply:
It's not used when publishing to latest 
```
 * Only used when publishing to the `dev` and `integration` npm channels
 * (see `getNewDevVersion()` and `getNewIntegrationVersion()`)
```

added this